src/probnum/random_variables/_normal.py

```diff
@@ -257,14 +258,30 @@ def __init__(
             entropy=entropy,
         )
 
-    @cached_property
+    @property
     def cov_cholesky(self) -> _ValueType:
         """Cholesky factor :math:`L` of the covariance
         :math:`\\operatorname{Cov}(X) =LL^\\top`."""
-        if self._compute_cov_cholesky is None:
-            raise NotImplementedError
 
-        return self._compute_cov_cholesky()
+        if not self.cov_cholesky_is_precomputed:
+            self.precompute_cov_cholesky()
+        return self._cov_cholesky_values
+
+    def precompute_cov_cholesky(self, damping_factor=COV_CHOLESKY_DAMPING):
+        """Overwrites values of cov_cholesky with new Cholesky factors (careful: in-
+        place operation!)."""
+        self._cov_cholesky_values = self._compute_cov_cholesky(
+            damping_factor=damping_factor
+        )
+
+    @property
+    def cov_cholesky_is_precomputed(self) -> bool:
+        """Whether the Cholesky factor of the covariance is readily available, which
+        happens if either cov_cholesky is specified during initialization or if the
+        property has been called."""
+        if self._cov_cholesky_values is None:
+            return False
+        return True
```

Comment:
this could be done in 1 line, but I personally prefer the readibility of this if-clause.